### PR TITLE
add test for store_admin adds photo to permitted store and cannot add…

### LINKED ISSUE #23: Business Admin adds photo
As a business admin, when I visit my business homepage and click "Upload photo to portfolio" I expect to be directed to a page where I can upload a photo and give it a name, description, and price. After filling in the information and clicking on "Upload" I expect to be taken to the new item show page.

### PR DESCRIPTION
**Pull request by mrjaimisra**
… photos to other stores

closes #23 

fix navbar layouts
hide button for adding photo to store to all users without store_admin permissions
add store has_many users and users belong_to store relationship
create routes for stores/edit, show, update
change urls for stores/show page to display store name instead of id
add store_id column to users table

### CODE REVIEW

File: app/views/stores/show.html.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+HELLO
```
Comment (mrjaimisra) [inline app/views/stores/show.html.erb]: placeholder... will change in forthcoming commits

File: app/controllers/stores/photos_controller.rb [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+
   def index
     @photos = @photographer.photos.paginate(page: params[:page]).order('created_at DESC')
   end
 
   def new
-    @photo = Photo.new
+    if current_user.store_id == current_store.id
+      photographer = Store.find_by(url: params[:photographer])
+      current_user && current_user.store_id == photographer.id
```
Comment (michael-reeves) [inline app/controllers/stores/photos_controller.rb]: what exactly is this line doing?

File: app/services/permission.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-class Permission
+  class Permission
```
Comment (michael-reeves) [inline app/services/permission.rb]: Don't need this space at beginning of lines

File: app/services/permission.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@ def allow?(controller, action)
     case
     when platform_admin?
       platform_admin_permissions
-    when store_admin?
+    when store_admin? #&& @user.store_id == @photographer.id
```
Comment (michael-reeves) [inline app/services/permission.rb]: if we're not using this, can we lose it?

File: app/services/permission.rb (hunk 3 above)
Comment (mrjaimisra) [inline app/services/permission.rb]: I'll be the judge of that

File: app/views/stores/show.html.erb (hunk 1 above)
Comment (michael-reeves) [inline app/views/stores/show.html.erb]: You can also use #TODO for this kind of thing

File: spec/features/store_admin_adds_photo_to_store_spec.rb [hunk 5]
```diff
@@ -1,20 +1,22 @@
 require "rails_helper"
 
 RSpec.feature "A store admin" do
-  let(:user)  { Fabricate(:user) }
-  let(:store) { Fabricate(:store) }
-  let(:photo) { Fabricate(:photo) }
+  attr_reader :store, :store_admin, :photo, :other_store
```
Comment (michael-reeves) [inline spec/features/store_admin_adds_photo_to_store_spec.rb]: Typically, we don't have readers in spec files?  It's fine to use @variables

File: app/controllers/stores/photos_controller.rb (hunk 2 above)
Comment (mrjaimisra) [inline app/controllers/stores/photos_controller.rb]: ...nothing :-1:

File: app/services/permission.rb (hunk 4 above)
Comment (mrjaimisra) [inline app/services/permission.rb]: gotcha. forgot to commit.

File: spec/features/store_admin_adds_photo_to_store_spec.rb (hunk 5 above)
Comment (mrjaimisra) [inline spec/features/store_admin_adds_photo_to_store_spec.rb]: I hate @ signs.

File: app/views/stores/show.html.erb (hunk 1 above)
Comment (mrjaimisra) [inline app/views/stores/show.html.erb]: nice! good call!

**Pull request merged**